src/pubvis.js

```diff
@@ -535,6 +539,37 @@ PUBVIS = function () {
                 //console.log( "keywords.length: " + keywords.length );
             }
 
+            exclude_chosen_keywords = function (keywords) {
+                if(exclude_chosen_keywords.length > 0) {
+                    var tmp_keywords = [];
+    
+                    for(var i = 0; i < keywords.length; i++) {
+                        if(!lockForPush) {tmp_keywords.push(keywords[i-1])}
```

Comment:
At this place, the `push` does not work fully as expected:
- in the first iteration it pushes `undefined` (which JavaScripts returns from `keywords[-1]`) 
- after the last iteration this code is not visited, thus the last keyword is not pushed.
Solution: move this line after the forEach (indicated below) and operate on the current index (`keywords[i]`)